### PR TITLE
Update installation to support new analysis tools

### PR DESCRIPTION
**Pull request by chameco**

### CODE REVIEW

File: .setup/INSTALL_SUBMITTY_HELPER.sh [hunk 1]
```diff
@@ -490,28 +490,23 @@ rm ${HWCRON_CRONTAB_FILE}
...
-#rsync -rtz ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT_AnalysisTools/bin/count_function  ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools/bin
-rsync -rtz ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT_AnalysisTools/lang/*              ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools/lang
-#rsync -rtz ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT_AnalysisTools/config/*            ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools/config
-rsync -rtz ${SUBMITTY_INSTALL_DIR}/GIT_CHECKOUT_AnalysisTools/bin/*               ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools/bin
+# compile the tools
+stack --allow-different-user --no-terminal --install-ghc --copy-bins build
+
+popd
+
+rm -rf ${SUBMITTY_INSTALL_DIR}/SubmittyAnalysisTools
```
Comment (bmcutler) [inline .setup/INSTALL_SUBMITTY_HELPER.sh]: Let's use an rsync (with or without the delete option) to minimize copying in the re-install case.
The install with 'clean' can be used to delete then re-copy the files if something has gone weird.

File: grading/execute.cpp [hunk 2]
```diff
@@ -63,9 +63,8 @@ bool system_program(const std::string &program, std::string &full_path_executabl
     { "tail",                    "/usr/bin/tail" },
 
     // Submitty Analysis Tools
-    { "submitty_count_token",    SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_token" },
-    { "submitty_count_node",     SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_node" },
-    { "submitty_count_function", SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_function" },
+    { "submitty_count",          SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/count" },
+    { "submitty_plagiarism",     SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/plagiarism" },
```
Comment (bmcutler) [inline grading/execute.cpp]: plagiarism doesn't need to be whitelisted.  it is not run within autograding.

File: grading/seccomp_functions.cpp [hunk 3]
```diff
@@ -128,14 +128,9 @@ int install_syscall_filter(bool is_32, const std::string &my_program, std::ofstr
   
   // ---------------------------------------------------------------
   // SUBMITTY ANALYSIS TOOLS
-  else if (my_program == SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_node" ||
-           my_program == SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_function" ||
-           my_program == SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/bin/count_token") {
+  else if (my_program == SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/count"
+		  || my_program == SUBMITTY_INSTALL_DIRECTORY+"/SubmittyAnalysisTools/plagiarism") {
```
Comment (bmcutler) [inline grading/seccomp_functions.cpp]: remove plagiarism

File: grading/seccomp_functions.cpp [hunk 4]
```diff
@@ -128,14 +128,9 @@ int install_syscall_filter(bool is_32, const std::string &my_program, std::ofstr
...
     categories = restricted_categories;
-    categories.insert("COMMUNICATIONS_AND_NETWORKING_SIGNALS");
```
Comment (bmcutler) [inline grading/seccomp_functions.cpp]: please replace these lines.  this is WIP  (add a "TODO" if you like)

File: sample_files/sample_assignment_config/cpp_goto_auto/config.json [hunk 5]
```diff
@@ -3,14 +3,15 @@
     "resource_limits" : {
         "RLIMIT_CPU" : 60,
         "RLIMIT_NPROC" : 100,
-        "RLIMIT_AS" : "RLIM_INFINITY"
+        "RLIMIT_AS" : "RLIM_INFINITY",
+	"RLIMIT_SIGPENDING" : 100
```
Comment (bmcutler) [inline sample_files/sample_assignment_config/cpp_goto_auto/config.json]: indentation?  looks weird on github (use spaces not tabs)

File: sample_files/sample_assignment_config/cpp_goto_auto/config.json [hunk 6]
```diff
@@ -3,14 +3,15 @@
...
     },
 
     "testcases" : [
         {
             "title" : "Check for goto and auto",
-            "command" : [ "submitty_count_token -l c goto *.cpp",
-                          "submitty_count_token -l c auto *.cpp" ],
+            "command" : [ "submitty_count token -l c Goto *.cpp",
```
Comment (bmcutler) [inline sample_files/sample_assignment_config/cpp_goto_auto/config.json]: is this supposed to be just submitty_count?

File: .travis.yml [hunk 7]
```diff
@@ -98,7 +98,7 @@ before_script:
   - psql -U postgres -c "CREATE ROLE hsdbu WITH SUPERUSER CREATEDB CREATEROLE LOGIN PASSWORD 'hsdbu'"
   - sudo ./.setup/travis/autograder.sh
   - pushd /usr/local/submitty/GIT_CHECKOUT_AnalysisTools
-  - travis_wait stack --allow-different-user --install-ghc --no-terminal --copy-bins build
+  - sudo travis_wait stack --allow-different-user --install-ghc --no-terminal --copy-bins build
```
Comment (MasterOdin) [inline .travis.yml]: `travis_wait sudo stack ...`

**Pull request merged**